src/main/java/com/avispl/symphony/dal/infrastructure/management/philips/huebridge/PhilipsHueDeviceCommunicator.java

```diff
@@ -0,0 +1,4123 @@
+	 * Populate cancel change button and edit is true
+	 *
+	 * @param stats the stats are list of statistics
+	 * @param advancedControllableProperties advancedControllableProperties is the list that store all controllable properties
+	 * @param propertyGroup the propertyGroup is name of group
+	 */
+	private void populateCancelChangeButton(Map<String, String> stats, List<AdvancedControllableProperty> advancedControllableProperties, String propertyGroup, boolean isCancelChangeButton) {
+		if (isCancelChangeButton) {
+			stats.put(propertyGroup + PhilipsConstant.HASH + PhilipsConstant.EDITED, PhilipsConstant.TRUE);
+			stats.put(propertyGroup + PhilipsConstant.HASH + PhilipsConstant.CANCEL_CHANGE, PhilipsConstant.EMPTY_STRING);
+			advancedControllableProperties.add(createButton(propertyGroup + PhilipsConstant.HASH + PhilipsConstant.CANCEL_CHANGE, PhilipsConstant.CANCEL, PhilipsConstant.CANCELING, 0));
+		}
+	}
+
+	/**
+	 * Send request to create/edit room and zone
+	 *
+	 * @param roomResponseData the roomResponseData is instance in RoomAndZoneResponse DTO
+	 * @param isRoomType the isRoomType is boolean value
+	 * @param isEditedRoomOrZone the isEditedRoomOrZone is boolean type if edit RoomOrZone then isEditedRoomOrZone = true and vice versa isEditedRoomOrZone = false
+	 */
+	private void sendRequestToCreateOrEditRoomAndZone(RoomAndZoneResponse roomResponseData, boolean isRoomType, boolean isEditedRoomOrZone) {
+		String request = PhilipsUtil.getMonitorURL(PhilipsURL.ZONES);
+		String name = PhilipsConstant.ZONE;
+		if (isRoomType) {
+			request = PhilipsUtil.getMonitorURL(PhilipsURL.ROOMS);
+			name = PhilipsConstant.ROOM;
+		}
+		try {
+			ResponseData responseData;
+			if (isEditedRoomOrZone) {
+				responseData = doPut(request + PhilipsConstant.SLASH + roomResponseData.getId(), roomResponseData.toString(), ResponseData.class);
+			} else {
+				responseData = doPost(request, roomResponseData.toString(), ResponseData.class);
+			}
+			if (responseData.getData() == null) {
+				throw new ResourceNotReachableException(String.format("Error while creating/editing %s: ", name) + Arrays.stream(responseData.getErrors()).map(ErrorsResponse::getDescription));
+			}
+		} catch (Exception e) {
+			throw new ResourceNotReachableException(String.format("Can't create/edit %s: ", name) + e.getMessage(), e);
+		}
+	}
+
+	/**
+	 * Send request to delete room and zone
+	 *
+	 * @param id the id is id of room and zone
+	 * @param isDeleteRoom the boolean type if isDeleteRoom = true is delete the room vice versa isDeleteRoom = false is to delete zone
+	 */
+	private void sendRequestToDeleteRoomAndZone(String id, boolean isDeleteRoom) {
+		String request = PhilipsUtil.getMonitorURL(PhilipsURL.ZONES);
+		if (isDeleteRoom) {
+			request = PhilipsUtil.getMonitorURL(PhilipsURL.ROOMS);
+		}
+		try {
+			doDelete(request + PhilipsConstant.SLASH + id);
+		} catch (Exception e) {
+			throw new ResourceNotReachableException("Can't delete room/zone: " + e.getMessage(), e);
```

Comment:
can you please include room/zone id into the exception message? It would be easier to understand at glance what is going on, plus exception message may be missing that if the error is unrelated

Reply:
I fixed it